File: src/main/java/net/hectus/neobb/turn/default_game/block/TDriedKelpBlock.java
package net.hectus.neobb.turn.default_game.block;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.WaterClazz;
import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import org.bukkit.block.Block;
import org.bukkit.potion.PotionEffectType;

import java.util.List;

public class TDriedKelpBlock extends BlockTurn implements BuffFunction, WaterClazz {
    public TDriedKelpBlock(NeoPlayer player) { super(player); }
    public TDriedKelpBlock(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 3;
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.Effect(Buff.BuffTarget.YOU, PotionEffectType.JUMP_BOOST, 2));
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/block/TDriedKelpBlock.kt
package net.hectus.neobb.turn.default_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.Effect
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.default_game.attribute.clazz.WaterClazz
import net.hectus.neobb.turn.default_game.attribute.function.BuffFunction
import org.bukkit.block.Block
import org.bukkit.potion.PotionEffectType

class TDriedKelpBlock(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), BuffFunction, WaterClazz {
    override val cost: Int = 3

    override fun buffs(): List<Buff<*>> {
        return listOf(Effect(PotionEffectType.JUMP_BOOST, 2))
    }
}